server/base/src/main/java/org/apache/accumulo/server/conf/ServerConfigurationFactory.java

```diff
@@ -140,4 +158,114 @@ public void connectionEvent() {
       // no-op. changes handled by prop store impl
     }
   }
+
+  private class ConfigRefreshRunner {
+    private static final long MIN_JITTER_DELAY = 1;
+
+    private static final long MAX_JITTER_DELAY = 23;
+    private final ScheduledFuture<?> refreshTaskFuture;
+
+    ConfigRefreshRunner() {
+
+      Runnable refreshTask = this::verifySnapshotVersions;
+      // staggering the initial delay prevents synchronization of Accumulo servers communicating
+      // with ZooKeeper for the sync process.
+      long randDelay = jitter(REFRESH_PERIOD_MINUTES / 4, REFRESH_PERIOD_MINUTES);
+
+      ScheduledThreadPoolExecutor executor = ThreadPools.getServerThreadPools()
+          .createScheduledExecutorService(4, "config-refresh", false);
+      refreshTaskFuture =
+          executor.scheduleWithFixedDelay(refreshTask, randDelay, REFRESH_PERIOD_MINUTES, MINUTES);
+    }
+
+    /**
+     * Check that the stored version in ZooKeeper matches the version held in the local snapshot.
+     * When a mismatch is detected, a change event is sent to the prop store which will cause a
+     * re-load. If the Zookeeper node has been deleted, the local cache entries are removed.
+     * <p>
+     * This method is designed to be called as a scheduled task, so it does not propagate exceptions
+     * other than interrupted Exceptions so the scheduled tasks will continue to run.
+     */
+
+    private synchronized void verifySnapshotVersions() {
```

Comment:
I think this synchronized isn't helping unless you make this method static.

Reply:
Fixed in ac14691996

Note - instead of creating a lock, I tried synchronizing on the AtomicBoolean - spotbugs didn't like it and flagged it as an error.